controller/eventController.js

```diff
@@ -0,0 +1,223 @@
+            message: 'Unauhorized.'
+        })
+    }
+
+    const event = new Event(req.body);
+
+    event.save().then(() => {
+        return res.status(HttpStatus.OK).json(event);
+    }).catch(err => {
+        const errorMessage = ValidationUtil.buildErrorMessage(err, 'create', 'event');
+        return res.status(HttpStatus.BAD_REQUEST).json({
+            status: 'Error',
+            message: errorMessage
+        });
+    });
+};
+
+const getAll = (req, res) => {
+    const user = req.decoded;
+    const companyId = user.companyID;
+
+    //All events can only be retrieved by an admin for the admin app
+    if (user && ValidationUtil.isUserAdmin(user.accessType)) {
+        //Retrieving only events for the company of the admin
+        Event.find({ companyID: companyId }).then((events) => {
+            if (events) {
+                return res.status(HttpStatus.OK).json(events);
+            } else {
+                return res.status(HttpStatus.NOT_FOUND).json({
+                    status: StatusEnum['ERROR'],
+                    message: 'Events not found.'
+                })
+            }
+        }).catch(err => {
+            return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+                status: StatusEnum['ERROR'],
+                message: 'Internal server error.'
+            });
+        });
+        //Regulal users can get all of the events where they are invited    
+    } else if (user && !ValidationUtil.isUserAdmin(user.accessType)) {
+        //TODO: Discuss better and faster solutions for this
+        Invite.find({ userID: new ObjectId(user._id) }).then(invites => {
+            if (!invites) {
+                return res.status(HttpStatus.NOT_FOUND).json({
+                    status: StatusEnum['ERROR'],
+                    message: 'No events for user.'
+                });
+            } else {
+                //First we extract the event id's from invites
+                let eventIDs = [];
+                invites.forEach(invite => {
+                    eventIDs.push(new ObjectId(invite.eventID));
+                });
+                //Now we find all the events for the event ids
+                Event.find({ '_id': { $in: eventIDs } }).then(events => {
+                    if (events) {
+                        return res.status(HttpStatus.OK).json(events);
+                    } else {
+                        return res.status(HttpStatus.NOT_FOUND).json({
+                            status: StatusEnum['ERROR'],
+                            message: 'No events for user.'
+                        });
+                    }
+                }).catch(err => {
+                    return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+                        status: StatusEnum['ERROR'],
+                        message: 'Internal server error'
+                    });
+                })
+            }
+        });
+    } else { //Case where the user is not authorized
```

Comment:
Just a suggestion. Maybe create a check first if `user` is present and if not return as it already is `INTERNAL_SERVER_ERROR`. This just makes the code cleaner since it doesn't have as many chained `if`, `else-if` and `else`s. Also makes further checks for `user` redundant (less checks => faster code).

Reply:
Moved the user check outside.